Tests/SwiftSyntaxBuilderTest/CommentsTests.swift

```diff
@@ -0,0 +1,282 @@
+    // Then
+    XCTAssertEqual(text, """
+      // This source file is part of the Swift.org open source project
+      // Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+
+      import SceneKit
+      """)
+  }
+  
+  func testSourceFileSingleLineComments() {
+    // Given
+    let source = SourceFile {
+      ImportDecl(path: "SceneKit")
+    }.lineComment("""
+    This source file is part of the Swift.org open source project
+    """)
+    // When
+    let syntax = source.buildSyntax(format: Format())
+    var text = ""
+    syntax.write(to: &text)
+    // Then
+    XCTAssertEqual(text, """
+      // This source file is part of the Swift.org open source project
+      
+      import SceneKit
+      """)
+  }
+  
+  
+  func testSourceFileCommentsWhenFirstNodeHasComment() {
+    // Given
+    let source = SourceFile {
+      ImportDecl(path: "SceneKit")
+        .lineComment("Create 3D games and add 3D content to apps using high-level scene descriptions.")
+    }.lineComment("""
+    This source file is part of the Swift.org open source project.
+    """)
+    // When
+    let syntax = source.buildSyntax(format: Format())
+    var text = ""
+    syntax.write(to: &text)
+    // Then
+    XCTAssertEqual(text, """
+      // This source file is part of the Swift.org open source project.
+      
+      // Create 3D games and add 3D content to apps using high-level scene descriptions.
+      import SceneKit
+      """)
+  }
+  
+  func testNestedCommentsIndentation() {
+    // Given
+    let source = SourceFile {
+      ImportDecl(path: "SceneKit")
+      ClassDecl(classOrActorKeyword: .class, identifier: "PlaneNode", membersBuilder: {
+        VariableDecl(.var, name: "planeGeometry", type: "SCNGeometry")
+          .lineComment("Geometry of plane, could be modified.")
+      })
+    }
+    // When
+    let syntax = source.buildSyntax(format: Format())
+    var text = ""
+    syntax.write(to: &text)
+    
+    // Then
+    XCTAssertEqual(text, """
+      
+      import SceneKit
+      class PlaneNode{
+          // Geometry of plane, could be modified.
+          var planeGeometry: SCNGeometry
+      }
+      """)
+  }
+  
+  func testMultipleLineCommentsIndentation() {
+    // Given
+    let source = SourceFile {
+      ImportDecl(path: "SceneKit")
+      ClassDecl(classOrActorKeyword: .class, identifier: "PlaneNode", membersBuilder: {
+        VariableDecl(.var, name: "planeGeometry", type: "SCNGeometry")
+          .lineComment("""
+    Plane node used to represent in a scene.
+    Each scene is supposed to have only one plane.
+    """
+          )
+      })
+    }
+    // When
+    let syntax = source.buildSyntax(format: Format())
+    var text = ""
+    syntax.write(to: &text)
+    print(text)
+    
+    // Then
+    XCTAssertEqual(text, """
+      
+      import SceneKit
+      class PlaneNode{
+          // Plane node used to represent in a scene.
+          // Each scene is supposed to have only one plane.
```

Comment:
It’s great that this works! Could you also add a similar test case for block comments?